pkg/systemd/quadlet/quadlet.go

```diff
@@ -540,10 +587,109 @@ func ConvertContainer(container *parser.UnitFile, isUser bool) (*parser.UnitFile
 	return service, nil
 }
 
+// Convert a quadlet network file (unit file with a Network group) to a systemd
+// service file (unit file with Service group) based on the options in the
+// Network group.
+// The original Network group is kept around as X-Network.
+func ConvertNetwork(network *parser.UnitFile, name string) (*parser.UnitFile, error) {
+	service := network.Dup()
+	service.Filename = replaceExtension(network.Filename, ".service", "", "-network")
+
+	if err := checkForUnknownKeys(network, NetworkGroup, supportedNetworkKeys); err != nil {
+		return nil, err
+	}
+
+	/* Rename old Volume group to x-Volume so that systemd ignores it */
+	service.RenameGroup(NetworkGroup, XNetworkGroup)
+
+	networkName := replaceExtension(name, "", "systemd-", "")
+
+	// Need the containers filesystem mounted to start podman
+	service.Add(UnitGroup, "RequiresMountsFor", "%t/containers")
+
+	podman := NewPodmanCmdline("network", "create")
+
+	if disableDns := network.LookupBoolean(NetworkGroup, KeyNetworkDisableDNS, false); disableDns {
+		podman.add("--disable-dns")
+	}
+
+	driver, ok := network.Lookup(NetworkGroup, KeyNetworkDriver)
+	if ok && len(driver) > 0 {
+		if _, ok := supportedNetworkDriversOptions[driver]; !ok {
```

Comment:
Why does this do any validation? Not just here but all of the code. This adds a lot of duplication.

IMO this is not maintainable. Nobody is going to remember to update this when we add a new network option. And alone the fact that I would even have to update this in two places is wrong.

Reply:
> The only thing I might still have to validate is the Subnets, Gateways, IPRanges count to make sure I can pass them correctly

That is fine for me but it should not try to validate the values, e.g. I currently work on adding network plugins which would allow you to use arbitrary driver names. The source of truth is in c/common/libnetwork, quadlet should restrict it further.

Reply:
Not sure I understood your last sentence
> quadlet should restrict it further

Let's make sure we're aligned. I'll remove all values validations. I will validate the counts of `subnet`, `gateway` and `iprange` so that they will make sense

Reply:
oh sorry it should have been `quadlet should NOT restrict it further`

> Let's make sure we're aligned. I'll remove all values validations. I will validate the counts of subnet, gateway and iprange so that they will make sense

correct

Reply:
Thanks for the clarification.

I've removed all values validations